### ISSUE TITLE
bug: exception handling with upstream changes

### ISSUE THREAD

**Issue by srevinsaju** [GabrielCastro/neon-serde#61]
following, possibly the API has changed. https://github.com/neon-bindings/neon/pull/542

Build fails with this error
```
2020-09-20T01:06:20.6410060Z error[E0061]: this function takes 3 arguments but 2 arguments were supplied
2020-09-20T01:06:20.6411547Z   --> /home/.cargo/registry/src/github.com-1ecc6299db9ec823/neon-serde-0.4.0/src/errors.rs:96:13
2020-09-20T01:06:20.6412101Z    |
2020-09-20T01:06:20.6412774Z 96 |             neon_runtime::error::throw_error_from_utf8(msg.as_ptr(), msg.len() as i32);
2020-09-20T01:06:20.6413482Z    |             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ ------------  ---------------- supplied 2 arguments
2020-09-20T01:06:20.6413770Z    |             |
2020-09-20T01:06:20.6414046Z    |             expected 3 arguments
2020-09-20T01:06:20.6414257Z 
2020-09-20T01:06:20.6656795Z error: aborting due to previous error

```

**Comment by srevinsaju** [comment]
For some reason, it works now, smh. I do not know why. See https://github.com/srevinsaju/element-appimage/actions which failed for days, but succeeded today.

**Comment by GabrielCastro** [comment]
@srevinsaju It looks like a breaking change in neon v0.4.1, for some reason it looks like your build went back to using an older v0.4.0 version

**Comment by srevinsaju** [comment]
Thanks, yes, and I do not know. The stable release fails, the continuous release builds from yesterday. 
Looks like its related to @jryans commit yesterday. Bumping `node-seshat` to the latest version, i.e, 2.1.0 -> 2.1.1 fixed the problem

https://github.com/vector-im/element-desktop/commit/161b024bf2f297f8ca0d1660fa107c49c28eff68


Here comes the right change that fixed my bug
https://github.com/matrix-org/seshat/commit/cbd3823665cf7fe69bbfc8a73af1a11dbe19542a

@poljar pinned `neon` to 0.4.0 and that fixed the problem.  I assumed its a bug over here. I apologize for the noise.
Thanks!

**Comment by poljar** [comment]
Neon introduced a breaking change in a patch release. The breaking change is this commit https://github.com/neon-bindings/neon/commit/a3503ee8242cc3f1db69af0c8341ae6565a7b0e0#diff-6fa50673a346bbd11536a64e3e825ca5L13.

Since neon-serde depends on neon and doesn't pin the version this caused breakage after a `cargo update`.

The bug isn't fixed, we just pinned the neon version to the latest working release.

**Comment by srevinsaju** [comment]
Yes I noticed. it requires a new positional argument `env: Env`. I guess I should reopen this issue. The author of `neon-serde` may choose to close the issue if its out of scope.

**Comment by goto-bus-stop** [comment]
Apologies from Neon—we started treating the neon_runtime module as internal during the N-API migration because it needed lots of these `Env` changes. I'll try to send a PR this week to update neon-serde's code for the changes. N-API requires an `Env` in order to throw an exception, so the `neon::Throw`-to-`neon_serde::Error` conversion probably needs to be handled a bit differently.

**Comment by poljar** [comment]
> Apologies from Neon—we started treating the neon_runtime module as internal during the N-API migration because it needed lots of these `Env` changes. I'll try to send a PR this week to update neon-serde's code for the changes. N-API requires an `Env` in order to throw an exception, so the `neon::Throw`-to-`neon_serde::Error` conversion probably needs to be handled a bit differently.

No problem, pinning versions is easy enough while everything gets updated.
